worksheet.rb

```diff
@@ -1,34 +1,112 @@
 # Step 1: Establish the layers
-
 # In this section of the file, as a series of comments,
 # create a list of the layers you identify.
 # Which layers are nested in each other?
 # Which layers of data "have" within it a different layer?
 # Which layers are "next" to each other?
 
-########################################################
-# Step 2: Assign a data structure to each layer
+# ANSWER: The outer layer is the one that contains the array. Within that outer layer,
+# lays the individual hashes layers (all 11 of them lay next to each other), but they count as one layer,
+# then within each "hash layer", there is a series of keys and their corresponding values; one of those keys (:date) contains
+# another nested layer, an array of length 3 that stores the date.
 
+########################################################################################################################################################################
+# Step 2: Assign a data structure to each layer
 # Copy your list from above, and in this section
 # determine what data structure each layer should have
 
-########################################################
-# Step 3: Make the data structure!
+# ANSWER: First layer: array; Second layer: hash, Third layer: array
 
+########################################################################################################################################################################
+# Step 3: Make the data structure!
 # Setup the entire data structure:
 # based off of the notes you have above, create the
 # and manually write in data presented in rides.csv
 # You should be copying and pasting the literal data
 # into this data structure, such as "DR0004"
 # and "3rd Feb 2016" and "RD0022"
 
-########################################################
-# Step 4: Total Driver's Earnings and Number of Rides
+# ANSWER: see code below
 
+arr_of_hashes = [
+  { driver_id: "DR0004", date: [02,03,2016], cost: 5, rider_id: "RD0022", rating: 5 },
+  { driver_id: "DR0001", date: [02,03,2016], cost: 10, rider_id: "RD0003", rating: 3 },
+  { driver_id: "DR0002", date: [02,03,2016], cost: 25, rider_id: "RD0073", rating: 5 },
+  { driver_id: "DR0001", date: [02,03,2016], cost: 30, rider_id: "RD0015", rating: 4 },
+  { driver_id: "DR0003", date: [02,04,2016], cost: 5, rider_id: "RD0066", rating: 5 },
+  { driver_id: "DR0004", date: [02,04,2016], cost: 10, rider_id: "RD0022", rating: 4 },
+  { driver_id: "DR0002", date: [02,04,2016], cost: 15, rider_id: "RD0013", rating: 1 },
+  { driver_id: "DR0003", date: [02,05,2016], cost: 50, rider_id: "RD0003", rating: 2 },
+  { driver_id: "DR0002",  date: [02,05,2016], cost: 35, rider_id: "RD0066", rating: 3 },
+  { driver_id: "DR0004", date: [02,05,2016], cost: 20, rider_id: "RD0073", rating: 5 },
+  { driver_id: "DR0001", date: [02,05,2016], cost: 45, rider_id: "RD0003", rating: 2 },
+  ]
```

Comment:
Consider how making the driver_id a key with the value being an array of hashes that represent each trip would simplify your code and make it a bit more flexible. For example:
```
drivers_and_rides = {
   DR001: [
      {
          date: ...
          cost: ...
          rider_id: ....
     },
    ....

```